core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberHandler.java

```diff
@@ -0,0 +1,62 @@
+package org.optaplanner.core.impl.domain.common.accessor.gizmo;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Type;
+import java.util.function.Consumer;
+
+import io.quarkus.gizmo.BytecodeCreator;
+import io.quarkus.gizmo.FieldDescriptor;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+
+interface GizmoMemberHandler {
+
+    /**
+     * Creates handler for a {@link Field}.
+     *
+     * @param declaringClass never null, class that declares the {@link Field} in question
+     * @param name never null, name of the field
+     * @param memberDescriptor never null, descriptor of the {@link Field} in question
+     * @param ignoreFinalChecks true if Quarkus will make the field non-final for us
+     * @return never null
+     */
+    static GizmoMemberHandler of(Class<?> declaringClass, String name, FieldDescriptor memberDescriptor,
```

Comment:
Soft suggestion: keep the parameter name consistent with its type (`methodDescriptor` / `fieldDescriptor`).